src/routes/messages/entities/message.entity.ts

```diff
@@ -7,6 +12,53 @@ export enum MessageStatus {
   Confirmed = 'CONFIRMED',
 }
 
+class TypedDataDomain {
+  @ApiPropertyOptional({ nullable: true }) name?: string;
+  @ApiPropertyOptional({ nullable: true }) version?: string;
+  @ApiPropertyOptional({ nullable: true }) chainId?: unknown;
+  @ApiPropertyOptional({ nullable: true }) verifyingContract?: string;
+  @ApiPropertyOptional({ nullable: true }) salt?: string | Array<number>;
+}
+
+class TypedDataTypes {
+  @ApiProperty() name: string;
+  @ApiProperty() type: string;
+
+  constructor(name: string, type: string) {
+    this.name = name;
+    this.type = type;
+  }
+}
+
+type TypedMessageTypes = Record<string, Array<TypedDataTypes>>;
+
+class EIP712TypedData {
```

Comment:
Nit: we should implement the types from `abitype`, e.g. `TypedData`. The argument types of `hashTypedData` will help guide.